src/diffusers/loaders/lora_base.py

```diff
@@ -904,3 +904,23 @@ def lora_scale(self) -> float:
         # property function that returns the lora scale which can be set at run time by the pipeline.
         # if _lora_scale has not been set, return 1
         return self._lora_scale if hasattr(self, "_lora_scale") else 1.0
+
+    def enable_lora_hotswap(self, **kwargs) -> None:
+        """Enables the possibility to hotswap LoRA adapters.
+
+        Calling this method is only required when hotswapping adapters and if the model is compiled or if the ranks of
+        the loaded adapters differ.
+
+        Args:
+            target_rank (`int`):
+                The highest rank among all the adapters that will be loaded.
+            check_compiled (`str`, *optional*, defaults to `"error"`):
```

Comment:
it is ok to have this argument here as long as user can easily diable it 

but I think it is more critical to add an argument similar this for unsupported cases that would ACTUALLY break the performance( e.g. `text_encoder` loras and different target layers), 

we should add check for these cases and throw a clear error message if user choose to error-out (otherwise we ignore or warn)